executor.go

```diff
@@ -25,6 +46,26 @@ func (e *executor) start() {
 			go func() {
 				defer wg.Done()
 
+				if e.maxRunningJobs != nil {
+					if !e.maxRunningJobs.TryAcquire(1) {
+
+						switch e.limitMode {
+						case RescheduleMode:
+							return
+						case WaitMode:
+							for {
```

Comment:
realized I need a way to stop this for loop if the scheduler is stopped - working on it

Reply:
oh, just realized we might need to take into consideration job removals as well right?